web/viewer.css

```diff
@@ -1371,208 +1077,650 @@ dialog :link {
+      color: var(--main-color);
+    }
+
+    &:is(:hover, :focus-visible) {
+      background-color: var(--button-hover-color);
+      color: var(--toggled-btn-color);
     }
   }
-}
 
-@page {
-  margin: 0;
+  &::after {
+    /* All matching images have a size of 16x16
+     * All relevant containers have a size of 28x28 */
+    position: absolute;
+    display: inline;
+    width: var(--icon-size);
+    height: var(--icon-size);
+
+    content: "";
+    background-color: var(--toolbar-icon-bg-color);
+    mask-size: cover;
+
+    inset-inline-end: 4px;
+    pointer-events: none;
+    mask-image: var(--toolbarButton-menuArrow-icon);
+  }
+
+  &:is(:hover, :focus-visible, :active)::after {
+    background-color: var(--toolbar-icon-hover-bg-color);
+  }
 }
 
-#printContainer {
-  display: none;
+#toolbarContainer {
+  --menuitem-height: calc(var(--toolbar-height) - 6px);
+
+  height: var(--toolbar-height);
+  padding: var(--toolbar-vertical-padding) var(--toolbar-horizontal-padding);
+  position: relative;
+  box-sizing: border-box;
+  font: message-box;
+  background-color: var(--toolbar-bg-color);
+  box-shadow: var(--toolbar-box-shadow);
+  border-bottom: var(--toolbar-border-bottom);
+
+  #toolbarViewer {
+    width: 100%;
+    height: 100%;
+    justify-content: space-between;
+
+    > * {
+      flex: none;
+    }
+
+    input {
+      font: message-box;
+    }
+
+    .toolbarButtonSpacer {
+      width: 30px;
+      display: block;
+      height: 1px;
+    }
+
+    #toolbarViewerLeft #numPages.toolbarLabel {
+      padding-inline-start: 3px;
+      flex: none;
+    }
+  }
+
+  #loadingBar {
+    /* Define these variables here, and not in :root, to avoid reflowing the
+       entire viewer when updating progress (see issue 15958). */
+    --progressBar-percent: 0%;
+    --progressBar-end-offset: 0;
+
+    position: absolute;
+    inset-inline: 0 var(--progressBar-end-offset);
+    height: 4px;
+    background-color: var(--progressBar-bg-color);
+    border-bottom: 1px solid var(--toolbar-border-color);
+    transition-property: inset-inline-start;
+    transition-duration: var(--sidebar-transition-duration);
+    transition-timing-function: var(--sidebar-transition-timing-function);
+
+    .progress {
+      position: absolute;
+      top: 0;
+      inset-inline-start: 0;
+      width: 100%;
+      transform: scaleX(var(--progressBar-percent));
+      transform-origin: calc(50% - 50% * var(--dir-factor)) 0;
+      height: 100%;
+      background-color: var(--progressBar-color);
+      overflow: hidden;
+      transition: transform 200ms;
+    }
+
+    &.indeterminate .progress {
+      transform: none;
+      background-color: var(--progressBar-blend-color);
+      transition: none;
+
+      .glimmer {
+        @keyframes progressIndeterminate {
+          0% {
+            transform: translateX(calc(-142px * var(--dir-factor)));
+          }
+
+          100% {
+            transform: translateX(0);
+          }
+        }
```

Comment:
It *seems* that `@keyframes` definitions perhaps cannot be nested like this, since moving it back to the "top level" as before fixes the animation.